src/components/button/Button.tsx

```diff
@@ -1,53 +1,147 @@
-import React, { MouseEventHandler } from "react";
+import React, { ButtonHTMLAttributes, forwardRef, ReactNode } from "react";
 import styled from "styled-components";
 
-export type ButtonProps = {
-  text?: string;
-  primary?: boolean;
+export interface ButtonProps
+  extends Omit<ButtonHTMLAttributes<HTMLButtonElement>, "title"> {
+  loading?: boolean;
```

Comment:
It looks like there is not any logic that pertains explicitly to loading. I would adovcate that we only include the `disabled` prop